pages/index.js

```diff
@@ -41,6 +44,38 @@ export default function Home({ total, today }) {
   const router = useRouter();
   const newsletter = router.query.newsletter;
 
+  const [randomProfile, setRandomProfile] = useState(null);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
```

Comment:
This would be better in `getStaticProps` similar to how the page does the other stats - then it will benefit from the caching

Reply:
I'm having a question,  Since the random profile should change on every page reload, using getStaticProps would mean that the random profile would remain the same until the next build, which is not the desired behavior. Correct? Or it should not change?.

Using the useEffect hook to fetch the random profile on the client-side when the component mounts. This way, we will get a new random profile on each page reload, as intended.? Since we are populating the homepage with a random Profile!

I'm gonna use the getStaticProps regardless. Just my Opinion or what i thought it should be. Thanks and i will be expecting feedback on this one

Reply:
> This would be better in `getStaticProps` similar to how the page does the other stats - then it will benefit from the caching

I'm having a question,  Since the random profile should change on every page reload, using getStaticProps would mean that the random profile would remain the same until the next build, which is not the desired behavior. Correct? Or it should not change?.

Using the useEffect hook to fetch the random profile on the client-side when the component mounts. This way, we will get a new random profile on each page reload, as intended.? Since we are populating the homepage with a random Profile!

I'm gonna use the getStaticProps regardless. Just my Opinion or what i thought it should be. Thanks and i will be expecting feedback on this one

Reply:
To keep the page performant it will be cache with the other stats, so it will change depending on the config

```js
  return {
    props: { total: totalStats, today: todayStats },
    revalidate: pageConfig.revalidateSeconds,
  };
```

Reply:
> To keep the page performant it will be cache with the other stats, so it will change depending on the config
> 
> ```js
>   return {
>     props: { total: totalStats, today: todayStats },
>     revalidate: pageConfig.revalidateSeconds,
>   };
> ```

Yes understood, done!